pkg/controller/identity/iamaccesskey/controller.go

```diff
@@ -0,0 +1,175 @@
+}
+
+type external struct {
+	client iam.AccessClient
+	kube   client.Client
+}
+
+func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.ExternalObservation, error) {
+	cr, ok := mgd.(*v1alpha1.IAMAccessKey)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errUnexpectedObject)
+	}
+
+	if meta.GetExternalName(cr) == "" {
+		return managed.ExternalObservation{}, nil
+	}
+
+	keys, err := e.client.ListAccessKeysRequest(&awsiam.ListAccessKeysInput{UserName: aws.String(cr.Spec.ForProvider.IAMUsername)}).Send(ctx)
+	if resource.Ignore(iam.IsErrorNotFound, err) != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errList)
+	}
+
+	if keys != nil {
+		for _, key := range keys.AccessKeyMetadata {
+			if aws.StringValue(key.AccessKeyId) == meta.GetExternalName(cr) {
+				cr.SetConditions(runtimev1alpha1.Available())
+
+				return managed.ExternalObservation{
+					ResourceExists:   true,
+					ResourceUpToDate: string(key.Status) == cr.Spec.ForProvider.Status,
+				}, nil
+			}
+		}
+	}
+
+	return managed.ExternalObservation{}, nil
+}
+
+func (e *external) Create(ctx context.Context, mgd resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mgd.(*v1alpha1.IAMAccessKey)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errUnexpectedObject)
+	}
+
+	cr.Status.SetConditions(runtimev1alpha1.Creating())
+
+	response, err := e.client.CreateAccessKeyRequest(&awsiam.CreateAccessKeyInput{UserName: aws.String(cr.Spec.ForProvider.IAMUsername)}).Send(ctx)
+	if err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errCreate)
+	}
+
+	var conn managed.ConnectionDetails = nil
+	if response != nil && response.AccessKey != nil {
+		conn = managed.ConnectionDetails{}
+		conn[runtimev1alpha1.ResourceCredentialsSecretUserKey] = []byte(aws.StringValue(response.AccessKey.AccessKeyId))
+		conn[runtimev1alpha1.ResourceCredentialsSecretPasswordKey] = []byte(aws.StringValue(response.AccessKey.SecretAccessKey))
+
+		cr.Spec.ForProvider.Status = string(response.AccessKey.Status)
+
+		meta.SetExternalName(cr, aws.StringValue(response.AccessKey.AccessKeyId))
+
+		err := e.kube.Update(ctx, cr)
+		if err != nil {
+			return managed.ExternalCreation{}, errors.Wrap(err, errKubeUpdateFailed)
+		}
+	}
+
+	return managed.ExternalCreation{ConnectionDetails: conn}, errors.Wrap(err, errCreate)
+}
+
+func (e *external) Update(ctx context.Context, mgd resource.Managed) (managed.ExternalUpdate, error) {
+	cr, ok := mgd.(*v1alpha1.IAMAccessKey)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errUnexpectedObject)
+	}
+
+	_, err := e.client.UpdateAccessKeyRequest(&awsiam.UpdateAccessKeyInput{
+		AccessKeyId: aws.String(meta.GetExternalName(cr)),
+		Status:      awsiam.StatusType(cr.Spec.ForProvider.Status),
+		UserName:    aws.String(cr.Spec.ForProvider.IAMUsername),
+	}).Send(ctx)
+
+	return managed.ExternalUpdate{}, errors.Wrap(resource.Ignore(iam.IsErrorNotFound, err), errUpdate)
```

Comment:
```suggestion
	return managed.ExternalUpdate{}, errors.Wrap(err, errUpdate)
```

If the resource does not exist, we should return an error here.